src/remote/blobs.go

```diff
@@ -274,40 +274,52 @@ func (c *Client) byteStreamDownloadName(digest *pb.Digest) string {
 // Data is not required.
 // The given function is a callback that receives a channel to send these blobs on; it
 // should close it when finished.
-func (c *Client) downloadBlobs(f func(ch chan<- *blob) error) error {
+func (c *Client) downloadBlobs(ctx context.Context, f func(ch chan<- *blob) error) error {
 	ch := make(chan *blob, 10)
+	done := make(chan struct{})
 	var g errgroup.Group
-	g.Go(func() error { return f(ch) })
+	go func() error {
+		defer close(done)
+		g.Go(func() error { return f(ch) })
```

Comment:
Do we need to start this in this goroutine? might be a little clearer to do it outside?

Reply:
It's a bit fiddly, but I want to make sure this is killed if we hit the context timeout - hence the `select` on `<-done` and `<-ctx.Done()`. Open to suggestions of a better way to do this though.